src/class/cdc/cdc_device.c

```diff
@@ -416,6 +416,9 @@ bool cdcd_xfer_cb(uint8_t rhport, uint8_t ep_addr, xfer_result_t result, uint32_
   //       Though maybe the baudrate is not really important !!!
   if ( ep_addr == p_cdc->ep_in )
   {
+    // invoke transmit callback to possibly refill tx fifo
+    if ( tud_cdc_tx_cb && !tu_fifo_full(&p_cdc->tx_ff) ) tud_cdc_tx_cb(itf);
```

Comment:
This is good idea, though I think we shouldn't invoke every transfer complete. Maybe invoke it when the TX FIFO is empty, which is just inside the below if() for Zero length packet. We should also rename it to `tud_cdc_tx_empty_cb()`